### PR TITLE
Use ESLint in Erizo Client.

### PR DESCRIPTION
**Pull request by jcague**
**Description**

Introduces ESLint and Airbnb style guide to check coding style in Erizo Client. +1000 linting issues showed up during the process so I had to modify a lot of lines, sorry. Most of them were related to wrong indentation, but there were many others.

[] It needs and includes Unit Tests

**Changes in Client or Server public APIs**

Not needed.

[] It includes documentation for these changes in `/doc`.

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
+    "extends": "airbnb",
```
Comment (lodoyun) [inline .eslintrc]: 👍

File: erizo_controller/erizoClient/src/Connection.js [hunk 2]
```diff
@@ -1,189 +1,183 @@
...
+          screenConfig.video.mediaSource = 'window' || 'screen';
+        } else {
+          screenConfig = {
+            audio: config.audio,
+            video: { mediaSource: 'window' || 'screen' },
+          };
+        }
+        if (navigator.mediaDevices && navigator.mediaDevices.getUserMedia) {
+          promise = navigator.mediaDevices.getUserMedia(screenConfig).then(callback);
                     // Google compressor complains about a func called catch
```
Comment (lodoyun) [inline erizo_controller/erizoClient/src/Connection.js]: If this isn't true anymore we could delete this comment

File: erizo_controller/erizoClient/src/utils/L.Resizer.js [hunk 3]
```diff
@@ -1,460 +1,462 @@
...
+} else {
+  window.attachEvent('onload', init);
+}
+
+  // Only used for the dirty checking, so the event callback count is limted
+  //  to max 1 call per fps per sensor.
+  // In combination with the event based resize sensor this saves cpu time,
+  // because the sensor is too fast and
+  // would generate too many unnecessary events.
+const requestAnimationFrame = window.requestAnimationFrame ||
```
Comment (lodoyun) [inline erizo_controller/erizoClient/src/utils/L.Resizer.js]: The google-closure-compiler is complaining here of a dublicate const declaration, not sure why.

File: erizo_controller/erizoClient/src/views/VideoPlayer.js [hunk 4]
```diff
@@ -1,179 +1,160 @@
...
-                    that.video.style.left = -((4 / 3) * height / 2 - width / 2) + 'px';
-                    that.video.style.top = '0px';
-
-                }
-            }
-        }
-
-        that.containerWidth = width;
-        that.containerHeight = height;
+Erizo.VideoPlayer = (spec) => {
```
Comment (lodoyun) [inline erizo_controller/erizoClient/src/views/VideoPlayer.js]: Consider `Erizo.VideoPlayer = function VideoPlayer(spec)`
Arrow functions cannot be constructors

File: erizo_controller/erizoClient/src/views/AudioPlayer.js [hunk 5]
```diff
@@ -1,111 +1,104 @@
...
  * AudioPlayer represents a Licode Audio component that shows either a local or a remote Audio.
  * Ex.: var player = AudioPlayer({id: id, stream: stream, elementID: elementID});
  * A AudioPlayer is also a View component.
  */
-var Erizo = Erizo || {};
-Erizo.AudioPlayer = function (spec) {
-    var that = Erizo.View({}),
-        onmouseover,
-        onmouseout;
+Erizo.AudioPlayer = (spec) => {
```
Comment (lodoyun) [inline erizo_controller/erizoClient/src/views/AudioPlayer.js]: Consider Erizo.AudioPlayer = function AudioPlayer(spec)

File: erizo_controller/erizoClient/src/utils/L.Resizer.js [hunk 6]
```diff
@@ -1,460 +1,462 @@
...
-                element = document.documentElement;
-            }
-            var fontSize = getComputedStyle(element, 'fontSize');
-            return parseFloat(fontSize) || 16;
-        }
+function getComputedStyle(computedElement, prop) {
+  if (computedElement.currentStyle) {
+    return computedElement.currentStyle[prop];
+  } else if (window.getComputedStyle) {
+    return window.getComputedStyle(computedElement, null).getPropertyValue(prop);
```
Comment (lodoyun) [inline erizo_controller/erizoClient/src/utils/L.Resizer.js]: I get an infinite loop here. Disabling it and making the other adjustments (constructor) works.

File: erizo_controller/erizoClient/src/views/Bar.js [hunk 7]
```diff
@@ -1,90 +1,90 @@
...
  * It contains a Speaker and an icon.
  * Every Bar is a View.
  * Ex.: var bar = Bar({elementID: element, id: id});
  */
-var Erizo = Erizo || {};
-Erizo.Bar = function (spec) {
-    var that = Erizo.View({}),
-        waiting,
-        show;
+Erizo.Bar = (spec) => {
```
Comment (lodoyun) [inline erizo_controller/erizoClient/src/views/Bar.js]: Consider `Erizo.Bar = function Bar(spec)`

File: erizo_controller/erizoClient/src/views/Speaker.js [hunk 8]
```diff
@@ -1,140 +1,137 @@
...
  * Ex.: var speaker = Speaker({elementID: element, media: mediaTag, id: id});
  */
-var Erizo = Erizo || {};
-Erizo.Speaker = function (spec) {
-    var that = Erizo.View({}),
-        show,
-        mute,
-        unmute,
-        lastVolume = 50;
+Erizo.Speaker = (spec) => {
```
Comment (lodoyun) [inline erizo_controller/erizoClient/src/views/Speaker.js]: Consider `Erizo.Speaker = Speaker(Spec) { `

**Comment by jcague** [comment]
@lodoyun I fixed all the issues, can you review it again please?

File: erizo_controller/erizoClient/src/utils/L.Resizer.js (hunk 3 above)
Comment (lodoyun) [inline erizo_controller/erizoClient/src/utils/L.Resizer.js]: This is still true, it does not seem to affect the functionality, though

**Pull request merged**